src/modules/ModuleSwap/store/swap.ts

```diff
@@ -45,8 +47,9 @@ function useSwap(input: Ref<null | NormalizedWeiInput>) {
   const scope = useParamScope(filteredKey, ({ props: { trade, tokenA, tokenB, amountFor }, dex }) => {
     const { state: prepareState, run: prepare } = useTask(
       async () => {
-        // 1. Approve amount of the tokenA
-        await dex.agent.approveAmount(tokenA.addr, tokenA.input)
+        // 1. Approve amount of the tokenA or max amount if expert mode is enabled
+        const amount = swapStore.expertMode ? new Wei(MAX_UINT256) : tokenA.input
+        await dex.agent.approveAmount(tokenA.addr, amount)
```

Comment:
I'd like to move "approval" logic to the `Swap` class. I know that I wrote this code... and I think it was not the best decision to make it in the Swap store. So, please add `expertMode` option to `SwapProps` and do approval there.

Also, `swapStore.expertMode` should be passed over `useParamScope` key - thus, if this option is updated, then outdated operation will be aborted and re-created.

Reply:
Done, check it, please